Src/World/ScriptWorld.h

```diff
@@ -31,18 +31,33 @@ class ScriptWorld {
         std::vector<NativeDefinition*> nativeDefs;
 
         EventDefinition* perTickEventDefinition;
+        EventDefinition* perEveryTickEventDefinition;
+        EventDefinition* perTickMenuEventDefinition;
+        EventDefinition* perLoadTickEventDefinition;
+
         EventDefinition* perFrameGameEventDefinition;
-        EventDefinition* perFrameMenuEventDefinition;
+        EventDefinition* perMenuFrameEventDefinition;
+        EventDefinition* perEveryFrameEventDefinition;
+        EventDefinition* perLoadFrameEventDefinition;
+
+        EventDefinition* resolutionChangedEventDefinition;
 
         std::vector<ScriptModule*> modules;
+        World* gameWorld;
 
     public:
-        ScriptWorld(World* world, GraphicsResources* gfxRes, Camera* camera, KeyBinds* keyBinds, MouseData* mouseData, PGE::IO* io, LocalizationManager* lm, PickableManager* pm, UIMesh* um,  Config* config, float timestep, BillboardManager* bm, ModelImageGenerator* mig);
+        ScriptWorld(World* world, GraphicsResources* gfxRes, Camera* camera, KeyBinds* keyBinds, MouseData* mouseData, PGE::IO* io, LocalizationManager* lm, PickableManager* pm, UIMesh* um,  Config* config, BillboardManager* bm, ModelImageGenerator* mig);
         ~ScriptWorld();
 
-        void update(float timeStep);
-        void drawGame(float interpolation);
-        void drawMenu(float interpolation);
+        void updateTick(uint32_t tick,float interp);
```

Comment:
The signature of this callback is ill-defined. The tick callback does not work with interpolation values, it works with timestep lengths.

Reply:
This is an interesting problem that I do not have a clean answer for.
Namely, how to manage animation frame advancement at the tick/framerate with regards to pausing.
This is a part of a bigger discussion that is worth having regarding how the timing systems are intended to work and what they need to do.